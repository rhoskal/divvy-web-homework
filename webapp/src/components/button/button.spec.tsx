import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Button } from "./Button";

describe("[Component] Button", () => {
  it("should handle onClick", () => {
    const onClick = jest.fn();

    render(<Button onClick={onClick}>Click Me</Button>);

    userEvent.click(screen.getByText("Click Me"));

    expect(onClick).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    const { container } = render(<Button>Click Me</Button>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
