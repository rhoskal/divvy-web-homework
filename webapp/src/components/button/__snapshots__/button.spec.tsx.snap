// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Button should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: rgba(0, 0, 0, 0.87);
  background-color: #90caf9;
  box-shadow: 0px 3px 1px -2px rgb(0 0 0 / 20%),0px 2px 2px 0px rgb(0 0 0 / 14%),0px 1px 5px 0px rgb(0 0 0 / 12%);
  padding: 6px 16px;
  min-width: 64px;
  font-size: 0.874rem;
  letter-spacing: 0.02857em;
  border-radius: 4px;
  font-weight: 500;
  font-family: "Roboto",sans-serif;
  text-transform: uppercase;
  border: 0;
  cursor: pointer;
  outline: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<button
  class="emotion-0"
>
  <span>
    Click Me
  </span>
</button>
`;
